src/hotspot/share/utilities/vmError.cpp

```diff
@@ -1382,18 +1390,34 @@ void VMError::print_vm_info(outputStream* st) {
   }
 #endif
 
+  // Take heap lock over both heap and GC printing so that information is
+  // consistent.
+  Heap_lock->lock();
```

Comment:
I would prefer using `MutexLocker` even though it changes indentation (of the source code).